model/credentials/db.go

```diff
@@ -0,0 +1,200 @@
+		DatabaseName:   settings.Database.DB,
+		CollectionName: Collection,
+	}
+}
+
+// getDepot returns the certificate depot connected to the database.
+func getDepot(ctx context.Context, env evergreen.Environment) (depot.Depot, error) {
+	return certdepot.NewMongoDBCertDepotWithClient(ctx, env.Client(), mongoConfig(env.Settings()))
+}
+
+func deleteIfExists(dpt depot.Depot, tag *depot.Tag) error {
+	if dpt.Check(tag) {
+		return errors.Wrap(dpt.Delete(tag), "problem deleting existing tag")
+	}
+	return nil
+}
+
+// SaveCredentials saves the credentials from the options for the given user
+// name. If the credentials already exist, this will overwrite the existing
+// credentials.
+func SaveCredentials(name string, creds *rpc.Credentials) error {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	defer cancel()
+
+	dpt, err := getDepot(ctx, evergreen.GetEnvironment())
+	if err != nil {
+		return errors.Wrap(err, "could not get depot")
+	}
+
+	if err := deleteIfExists(dpt, depot.PrivKeyTag(name)); err != nil {
+		return errors.Wrap(err, "problem deleting existing key")
+	}
+	if err := deleteIfExists(dpt, depot.CrtTag(name)); err != nil {
+		return errors.Wrap(err, "problem deleting existing certificate")
+	}
+
+	if err := dpt.Put(depot.PrivKeyTag(name), creds.Key); err != nil {
+		return errors.Wrap(err, "problem saving key")
+	}
+	if err := dpt.Put(depot.CrtTag(name), creds.Cert); err != nil {
+		return errors.Wrap(err, "problem saving certificate")
+	}
+	return nil
+}
+
+// GenerateInMemory generates the credentials without storing them in the
+// database.
+func GenerateInMemory(name string) (*rpc.Credentials, error) {
+	opts := certdepot.CertificateOptions{
+		CA:         CAName,
+		CommonName: name,
+		Host:       name,
+		Expires:    defaultExpiration,
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	defer cancel()
+
+	dpt, err := getDepot(ctx, evergreen.GetEnvironment())
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get depot")
+	}
+
+	pemCACrt, err := dpt.Get(depot.CrtTag(CAName))
+	if err != nil {
+		return nil, errors.Wrap(err, "problem getting CA certificate")
+	}
+
+	_, key, err := opts.CertRequestInMemory(dpt)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem making certificate request and key")
+	}
+
+	pemKey, err := key.ExportPrivate()
+	if err != nil {
+		return nil, errors.Wrap(err, "problem exporting key")
+	}
+
+	crt, err := opts.SignInMemory(dpt)
+	if err != nil {
+		return nil, errors.Wrap(err, "problem signing certificate request")
+	}
+
+	pemCrt, err := crt.Export()
+	if err != nil {
+		return nil, errors.Wrap(err, "problem exporting certificate")
+	}
+
+	return rpc.NewCredentials(pemCACrt, pemCrt, pemKey)
+}
+
+// FindById gets the credentials for the given name.
+func FindById(name string) (*rpc.Credentials, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultConnectionTimeout)
+	defer cancel()
```

Comment:
these operations should either get a context, or get the global context that you can get from the environment.